test/jdk/java/awt/Component/ComponentRedrawnTest.java

```diff
@@ -0,0 +1,203 @@
+
+        componentPanel = new Panel();
+        componentPanel.setLayout(null);
+        componentPanel.setBackground(Color.green);
+
+        awtButton = new Button("AWT Button") {
+            @Override
+            public void paint(Graphics g) {
+                super.paint(g);
+                awtPainted.incrementAndGet();
+            }
+        };
+
+        awtButton.setBounds(0, 0, 330, 100);
+        componentPanel.add(awtButton);
+
+        JButton swingButton = new JButton("Swing JButton") {
+            @Override
+            public void paint(Graphics g) {
+                super.paint(g);
+                swingPainted.incrementAndGet();
+            }
+        };
+
+        swingButton.setBounds(0, 100, 330, 100);
+        componentPanel.add(swingButton);
+        frame.add(componentPanel, BorderLayout.CENTER);
+        buttonRemove = new Button("remove");
+        buttonRemove.addActionListener(ae -> buttonClicked(ae));
+
+        buttonAdd = new Button("add");
+        buttonAdd.addActionListener(ae -> buttonClicked(ae));
+
+        Panel controlPanel = new Panel();
+        controlPanel.setLayout(new BorderLayout());
+        controlPanel.add(buttonRemove, BorderLayout.NORTH);
+        controlPanel.add(buttonAdd, BorderLayout.SOUTH);
+
+        frame.add(controlPanel, BorderLayout.SOUTH);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void buttonClicked(ActionEvent ae) {
+        if (ae.getSource() == buttonRemove) {
+            frame.remove(componentPanel);
+        } else if (ae.getSource() == buttonAdd) {
+            frame.add(componentPanel);
+        }
+        frame.invalidate();
+        frame.validate();
+    }
+
+    private static void runTest() throws AWTException {
+        try {
+            EventQueue.invokeAndWait(() -> createGUI());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        robot = new Robot();
+        robot.setAutoDelay(500);
+        awtPainted.set(0);
+        swingPainted.set(0);
+
+        try {
+            EventQueue.invokeAndWait(() -> {
+                x = awtButton.getLocationOnScreen().x
+                    + awtButton.getSize().width / 2;
+                y = awtButton.getLocationOnScreen().y
+                    + awtButton.getSize().height / 2;
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        robot.mouseMove(x, y);
+        robot.waitForIdle();
+        robot.mousePress(MouseEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(MouseEvent.BUTTON1_DOWN_MASK);
+
+        try {
+            EventQueue.invokeAndWait(() -> {
+                x = buttonRemove.getLocationOnScreen().x
+                    + buttonRemove.getSize().width / 2;
+                y = buttonRemove.getLocationOnScreen().y
+                    + buttonRemove.getSize().height / 2;
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
It is better to rethrow this exception and do not skip it, since the later part of the code will not make any sense after this exception.